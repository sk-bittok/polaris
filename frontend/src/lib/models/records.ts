export interface ProductionRecord {
	id: number;
	livestockName: string;
	animalPid: string;
	productType: string;
	unit: string;
	quantity: string;
	notes?: string;
	quality?: string;
	recordDate?: Date;
	createdAt: Date;
	createdBy: string;
	createdByName: string;
	updatedAt: Date;
}

export interface NewProductionRecord {
	id: number;
	animal_pid: string;
	organisation_pid: string;
	product_type: string;
	quantity: number;
	unit: string;
	record_date?: string;
	quality?: string;
	notes?: string;
	created_by: string;
	created_at: Date;
	updated_at: Date;
}

export interface HealthRecordResponse {
	id: number;
	organisationPid: string;
	organisationName: string;
	animalName: string;
	animalPid: string;
	animalTagId: string;
	recordType: string;
	recordDate: Date;
	description: string;
	treatment: string;
	medicine?: string;
	dosage?: string;
	cost?: number;
	notes?: string;
	performedBy?: string;
	createdBy: string;
	createdByName: string;
	createdAt: Date;
	updatedAt: Date;
}

export interface HealthRecord {
	id: number;
	organisation_pid: string;
	animal_pid: string;
	record_type: string;
	record_date: Date;
	description: string;
	treatment: string;
	medicine?: string;
	dosage?: string;
	cost?: number;
	notes?: string;
	performed_by?: string;
	created_by: string;
	created_at: Date;
	updated_at: Date;
}

export interface WeightRecord {
	id: number;
	organisation_pid: string;
	animal_pid: string;
	mass: number;
	record_date: string;
	notes?: string;
	created_by: string;
	created_at: Date;
	updated_at: Date;
}

export interface WeightRecordResponse {
	id: number;
	organisationPid: string;
	organisationName: string;
	animalPid: string;
	animalName: string;
	animalTagId: string;
	mass: number;
	recordDate: string;
	notes?: string;
	createdBy: string;
	createdByName: string;
	created_at: Date;
	updated_at: Date;
}
