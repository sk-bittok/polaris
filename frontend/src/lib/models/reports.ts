interface LivestockSummary {
	pid: string;
	organisationPid: string;
	id: number;
	total: number;
	males: number;
	females: number;
	unkownGender: number;
	active: number;
	transferred: number;
	sold: number;
	deceased: number;
	species: number;
	breeds: number;
	totalPurchasedValue: number;
	createdAt: Date;
	updatedAt: Date;
}

export type { LivestockSummary };
