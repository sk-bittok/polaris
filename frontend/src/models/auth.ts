import { z } from "zod";

export enum SubscriptionType {
    Basic = "basic",
    Business = "business",
    Enterprise = "enterprise",
}

export const companyFormSchema = z.object({
    companyName: z.string().min(2, { message: "Name requires 2 characters" })
        .max(200, { message: "Name is too long" }),
    companyAddress: z.string().max(200, { message: "Address is too long" })
        .nullable(),
    companyPhone: z.string().min(10, { message: "Phone is too short" })
        .max(12, { message: "Phone is too long" }).nullable(),
    companyEmail: z.string().email({ message: "Invalid email" }).nullable(),
    subscriptionType: z.enum(["basic", "business", "enterprise"], {
        message: "Basic, business and enterprise only",
    }),
});

export const userFormSchema = z.object({
    email: z.string().email({ message: "Invalid email" }),
    firstName: z.string().min(2, { message: "Name is too short" })
        .max(50, { message: "Name is too long" }),
    lastName: z.string().min(2, { message: "Name is too short" }).max(50, {
        message: "Name is too long",
    }),
    password: z.string().min(8, { message: "Password requires 8 characters" })
        .max(48, { message: "Password must be under 48 chars" }),
    confirmPassword: z.string(),
})
    .refine((data) => data.confirmPassword === data.password, {
        message: "Passwords must match",
        path: ["confirmPassword"],
    });

export const loginFormSchema = z.object({
    email: z.string().email({ message: "Invalid email" }),
    password: z.string().min(2, { message: "Password too short" })
        .max(100, { message: "Password too long" }),
});

export type CompanyForm = z.infer<typeof companyFormSchema>;
export type UserForm = z.infer<typeof userFormSchema>;

export type RegisterOrgAndUser = {
    organisation: CompanyForm;
    user: UserForm;
};

export type LoginForm = z.infer<typeof loginFormSchema>;
