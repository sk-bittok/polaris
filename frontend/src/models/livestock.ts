
export interface Breed {
  id: number;
  organisationPid?: string;
  specie: Category;
  name: string;
  maleWeightRange?: string;
  femaleWeightRange?: string;
  gestationPeriod?: string;
  description?: string;
  createdAt: string;
  isSystemDefined: boolean;
}

export enum Category {
  Cattle = "cattle",
  Sheep = "sheep",
  Goat = "goats",
  Chicken = "chicken",
  Pig = "pigs"
}

export enum Gender {
  Male = "male",
  Female = "female",
  Unkown = "unkown",
}

export enum Status {
  Active = "active",
  Sold = "sold",
  Deceased = "deceased",
  Transferred = "transferred"
}

export interface RegisterLivestock {
  tagId: string;
  name: string;
  gender: string;
  status: string;
  breed: string;
  specie: string;
  dateOfBirth?: string | Date | null;
  weightAtBirth?: number | null;
  maleParentId?: string | null;
  femaleParentId?: string | null;
  purchasePrice?: number | null;
  purchaseDate?: string | Date | null;
  currentWeight?: number | null;
  notes?: string | null;
};

export interface UpdateLivestock {
  tagId?: string | null;
  name?: string | null;
  gender?: string | null;
  status?: string | null;
  breed?: string | null;
  specie?: string | null;
  dateOfBirth?: string | Date | null;
  weightAtBirth?: number | null;
  maleParentId?: string | null;
  femaleParentId?: string | null;
  purchasePrice?: number | null;
  purchaseDate?: string | Date | null;
  currentWeight?: number | null;
  notes?: string | null;
};

export interface Livestock {
  id: number;
  pid: string;
  organisationName: string;
  tagId: string;
  name: string;
  breedName: string;
  specieName: Category;
  dateOfBirth?: Date;
  gender: Gender;
  parentMaleName?: string;
  parentFemaleName?: string;
  status: Status;
  parentMaleTagId?: string;
  parentFemaleTagId?: string;
  purchaseDate?:  Date;
  purchasePrice?: string;
  weightAtBirth?: string;
  currentWeight?: string;
  notes?: string;
  createdByName: string;
  createdAt: Date;
  updatedAt: Date;
}

