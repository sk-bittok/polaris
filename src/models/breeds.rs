#![allow(clippy::missing_errors_doc)]

use std::borrow::Cow;

use chrono::{DateTime, FixedOffset};
use serde::{Deserialize, Serialize};
use sqlx::{Encode, Executor, Postgres, prelude::FromRow};
use uuid::Uuid;

use super::{ModelError, ModelResult};

#[derive(Debug, Clone, Deserialize)]
pub struct BreedQuery {
    specie: Option<String>,
    name: Option<String>,
}

#[derive(Debug, Deserialize, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RegisterBreed<'a> {
    pub(crate) specie: Cow<'a, str>,
    pub(crate) name: Cow<'a, str>,
    pub(crate) description: Option<Cow<'a, str>>,
    pub(crate) typical_male_weight_range: Option<Cow<'a, str>>,
    pub(crate) typical_female_weight_range: Option<Cow<'a, str>>,
    pub(crate) typical_gestation_period: Option<Cow<'a, str>>,
}

#[derive(Debug, Deserialize, Serialize, Clone, FromRow, Encode)]
pub struct Breed {
    pub(crate) id: i32,
    pub(crate) specie: String,
    pub(crate) name: String,
    pub(crate) description: Option<String>,
    pub(crate) typical_male_weight_range: Option<String>,
    pub(crate) typical_female_weight_range: Option<String>,
    pub(crate) typical_gestation_period: Option<String>,
    pub(crate) created_at: DateTime<FixedOffset>,
    pub(crate) organisation_pid: Option<Uuid>,
    pub(crate) is_system_defined: bool,
}

impl Breed {
    pub async fn find_by_id<'e, C>(db: C, id: i32, org_pid: Uuid) -> ModelResult<Self>
    where
        C: Executor<'e, Database = Postgres>,
    {
        sqlx::query_as::<_, Self>("SELECT * FROM breeds WHERE id = $1 AND (is_system_defined = TRUE OR organisation_pid = $2)")
            .bind(id)
            .bind(org_pid)
            .fetch_optional(db)
            .await?
            .ok_or_else(|| ModelError::EntityNotFound)
    }

    pub async fn find_by_all<'e, C>(
        db: C,
        org_pid: Uuid,
        conditions: &BreedQuery,
    ) -> ModelResult<Vec<Self>>
    where
        C: Executor<'e, Database = Postgres>,
    {
        let mut query =
            sqlx::query_as::<_, Self>("SELECT * FROM breeds WHERE  (is_system_defined = TRUE OR organisation_pid = $1) ORDER BY name")
                .bind(org_pid);

        if let Some(name) = &conditions.name {
            query = sqlx::query_as::<_, Self>(
                "SELECT * FROM breeds WHERE (is_system_defined = TRUE OR organisation_pid = $1) AND name LIKE $2",
            )
            .bind(org_pid)
            .bind(format!("%{name}%"));
        }

        if let Some(specie) = &conditions.specie {
            query = sqlx::query_as::<_, Self>(
                "SELECT * FROM breeds WHERE organisation_pid = $1 AND specie LIKE $2",
            )
            .bind(org_pid)
            .bind(format!("%{specie}%",));
        }

        query.fetch_all(db).await.map_err(Into::into)
    }

    pub async fn create<'e, C>(db: C, org_pid: Uuid, dto: &RegisterBreed<'_>) -> ModelResult<Self>
    where
        C: Executor<'e, Database = Postgres>,
    {
        let query = sqlx::query_as::<_, Self>(
            "INSERT INTO breeds (specie, name, typical_male_weight_range, typical_female_weight_range, typical_gestation_period, organisation_pid, description)
                VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *
            ")
        .bind(&dto.specie)
        .bind(&dto.name)
        .bind(&dto.typical_male_weight_range)
        .bind(&dto.typical_female_weight_range)
        .bind(&dto.typical_gestation_period)
        .bind(org_pid)
        .bind(&dto.description)
        .fetch_one(db)
        .await?;

        Ok(query)
    }
}
