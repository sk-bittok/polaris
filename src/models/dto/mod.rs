pub mod auth;

pub use self::auth::*;
