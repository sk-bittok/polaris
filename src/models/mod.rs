pub mod animals;
pub mod breeds;
pub mod dto;
pub mod enums;
pub mod errors;
pub mod health_records;
pub mod livestock;
pub mod orgs;
pub mod production_records;
pub mod roles;
pub mod species;
pub mod users;
pub mod weight;

pub use self::errors::{ModelError, ModelResult};
