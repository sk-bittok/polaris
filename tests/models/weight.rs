use std::borrow::Cow;

use chrono::NaiveDate;
use insta::{assert_debug_snapshot, with_settings, Settings};
use polaris::models::{dto::records::NewWeightRecord, weight::{WeightQuery, WeightRecord}};
use rstest::rstest;
use rust_decimal::Decimal;
use serial_test::serial;
use uuid::Uuid;

macro_rules! configure_insta {
    ($(expr:expr),*) => {
        let mut settings = Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_path("snapshots/records");
        settings.set_snapshot_suffix("weight");
        let _guard = settings.bind_to_scope();
    };
}

#[rstest]
#[case(
    "can_find_all",
    WeightQuery {
        animal: None,
        mass: None,
    },
    Uuid::parse_str("9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0").unwrap()
)]
#[case(
    "can_find_all_animal_query",
    WeightQuery {
        animal: Some( Uuid::parse_str("f6417c11-d817-4626-9e8d-c68a44002d4b").unwrap()),
        mass: None,
    },
    Uuid::parse_str("9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0").unwrap()
)]
#[case(
    "can_find_all_mass_query",
    WeightQuery {
        animal:None, 
        mass: Some(Decimal::new(70, 0)),
    },
    Uuid::parse_str("4a0f3af9-e56e-4e21-8f3a-f9e56efe215b").unwrap()
)]
#[tokio::test]
#[serial]
async fn can_find_all(
    #[case] test_name: &str,
    #[case] conditions: WeightQuery,
    #[case] org_pid: Uuid,
) {
    configure_insta!();

    let ctx = crate::boot_test().await.unwrap();
    crate::seed_data(&ctx.db).await.unwrap();

    let results = WeightRecord::find_all(&ctx.db, org_pid, &conditions).await;

    assert_debug_snapshot!(test_name, results);
}


#[tokio::test]
#[serial]
async fn can_find_by_id() {
    configure_insta!();

    let ctx = crate::boot_test().await.unwrap();
    crate::seed_data(&ctx.db).await.unwrap();

    let org_pid =  Uuid::parse_str("9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0").unwrap();

    let results = WeightRecord::find_by_id(&ctx.db, org_pid, 115).await;

    assert_debug_snapshot!(results);
}


#[tokio::test]
#[serial]
async fn can_delete_one() {
    configure_insta!();

    let ctx = crate::boot_test().await.unwrap();
    crate::seed_data(&ctx.db).await.unwrap();

    let org_pid =  Uuid::parse_str("9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0").unwrap();

    let results = WeightRecord::delete_by_id(&ctx.db, org_pid, 115).await;

    assert_debug_snapshot!(results);
}

#[tokio::test]
#[serial]
async fn can_create_one() {
    configure_insta!();

    let ctx = crate::boot_test().await.unwrap();
    crate::seed_data(&ctx.db).await.unwrap();

    let org_pid =  Uuid::parse_str("9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0").unwrap();
    let user_pid =  Uuid::parse_str("bd6f7c26-d2c9-487e-b837-8f77be468033").unwrap();

    let params = NewWeightRecord {
        tag_id: Cow::Borrowed("AC007"),
        record_date: NaiveDate::parse_from_str("2025-04-22", "%Y-%m-%d").unwrap(),
        mass: 48800,
        notes: None
    };

    let results = WeightRecord::create(&ctx.db, &params, org_pid, user_pid).await;

    with_settings!({
        filters => {
            let mut filters = crate::cleanup_uuid().to_vec();
            filters.extend(crate::cleanup_date().to_vec());
            filters.extend(crate::cleanup_int().to_vec());
            filters
        }
    }, {
        assert_debug_snapshot!(results);
    })
}
