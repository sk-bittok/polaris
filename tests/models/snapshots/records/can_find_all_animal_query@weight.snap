---
source: tests/models/weight.rs
assertion_line: 57
expression: results
---
Ok(
    [
        WeightResponse {
            id: 101,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 32.00,
            record_date: 2024-04-10,
            notes: Some(
                "Birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 102,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 64.00,
            record_date: 2024-05-10,
            notes: Some(
                "Month after birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 103,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 100.00,
            record_date: 2024-06-10,
            notes: Some(
                "Two month after birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 104,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 136.00,
            record_date: 2024-07-10,
            notes: Some(
                "Three month after birth weight weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 105,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 168.00,
            record_date: 2024-08-10,
            notes: Some(
                "Four months weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 106,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 202.00,
            record_date: 2024-09-10,
            notes: Some(
                "Five month after birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 107,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 238.00,
            record_date: 2024-10-11,
            notes: Some(
                "Six monts after birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 108,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 276.00,
            record_date: 2024-11-10,
            notes: Some(
                "Seven months after birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 109,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 300.00,
            record_date: 2024-12-10,
            notes: Some(
                "Eight months after birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 110,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 342.00,
            record_date: 2025-01-10,
            notes: Some(
                "Nine months after birth weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 111,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 384.00,
            record_date: 2025-02-10,
            notes: Some(
                "Weaning weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 112,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 426.00,
            record_date: 2025-03-10,
            notes: Some(
                "One month post weaning weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 113,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 460.00,
            record_date: 2025-04-10,
            notes: Some(
                "Yearling weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 114,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 500.00,
            record_date: 2025-05-16,
            notes: Some(
                "Four hundred day weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2025-05-10T07:15:00+00:00,
            updated_at: 2025-05-10T07:15:00+00:00,
        },
        WeightResponse {
            id: 115,
            organisation_pid: 9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0,
            organisation_name: "Acme Corp",
            animal_pid: f6417c11-d817-4626-9e8d-c68a44002d4b,
            animal_name: "Rose",
            animal_tag_id: "AC007",
            mass: 545.00,
            record_date: 2025-06-10,
            notes: Some(
                "Two monts post yearling weight",
            ),
            created_by: bd6f7c26-d2c9-487e-b837-8f77be468033,
            created_by_name: "John Doe",
            created_at: 2024-04-10T07:15:00+00:00,
            updated_at: 2024-04-10T07:15:00+00:00,
        },
    ],
)
